src/main.rs

```diff
@@ -52,118 +53,145 @@ struct Args {
-                }
-                None => None,
-            };
-
-            let handler_result = route
-                .runner
-                .run(&runner::build_wasm_input(&req, body_str, store), vars)
-                .unwrap_or_else(|_| {
-                    WasmOutput::new(
-                        "<p>There was an error running this function</p>",
-                        HashMap::from([("content-type".to_string(), "text/html".to_string())]),
-                        StatusCode::SERVICE_UNAVAILABLE.as_u16(),
-                        HashMap::new(),
-                    )
-                });
-
-            let mut builder = HttpResponse::build(
-                StatusCode::from_u16(handler_result.status).unwrap_or(StatusCode::OK),
-            );
-            // Default content type
-            builder.insert_header(("Content-Type", "text/html"));
-
-            for (key, val) in handler_result.headers.iter() {
-                // Note that QuickJS is replacing the "-" character
-                // with "_" on property keys. Here, we rollback it
-                builder.insert_header((key.replace('_', "-").as_str(), val.as_str()));
+        // Let's continue
+        let body_str = String::from_utf8(body.to_vec()).unwrap_or_else(|_| String::from(""));
+
+        // Init from configuration
+        let empty_hash = HashMap::new();
+        let mut vars = &empty_hash;
+        let mut kv_namespace = None;
+
+        match &route.config {
+            Some(config) => {
+                kv_namespace = config.data_kv_namespace();
+                vars = &config.vars;
             }
+            None => {}
+        };
 
-            // Write to the state if required
-            if kv_namespace.is_some() {
-                data_connectors
-                    .write()
-                    .unwrap()
-                    .kv
-                    .replace_store(&kv_namespace.unwrap(), &handler_result.kv)
+        let store = match &kv_namespace {
+            Some(namespace) => {
+                let connector = data_connectors.read().unwrap();
+                let kv_store = connector.kv.find_store(namespace);
+
+                kv_store.map(|store| store.clone())
             }
+            None => None,
+        };
 
-            result = match handler_result.body() {
-                Ok(res) => builder.body(res),
-                Err(_) => {
-                    HttpResponse::ServiceUnavailable().body("There was an error running the worker")
-                }
+        let handler_result = route
+            .runner
+            .run(&req, body_str, store, vars)
+            .unwrap_or_else(|_| WasmOutput::failed());
+
+        let mut builder = HttpResponse::build(
+            StatusCode::from_u16(handler_result.status).unwrap_or(StatusCode::OK),
+        );
+        // Default content type
+        builder.insert_header(("Content-Type", "text/html"));
+
+        for (key, val) in handler_result.headers.iter() {
```

Comment:
Is iter() really needed? I think the preferred style is using a reference instead of explicit `.iter()`:
https://rust-lang.github.io/rust-clippy/master/index.html#explicit_iter_loop.

```suggestion
        for (key, val) in &handler_result.headers {
```